### PR TITLE
[Soh Jun Qi] Renaming of AddressBook to DietLah

### LINKED ISSUE #117: Removal of unused AB3 Code
As DietLAH! builds upon existing AB3 code, there has been code that is no longer used in the application. These codes need to be removed.

### LINKED ISSUE #170: [PE-D] Addressbook sample storage in date file
The sample for addressbook still exist inside the date file even though it is not relevant anymore.

-------------
Labels: `severity.Low` `type.FunctionalityBug`
original: NBH99/ped#3

### PR DESCRIPTION
**Pull request by sjq-sohjunqi**
Certain parts of the code was named AddressBook due to it being part of pre-existing code.

Let's rename all AddressBook instances into DietLah.

### CODE REVIEW

File: src/test/java/seedu/address/storage/JsonDietLahStorageTest.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        // Save and read without specifying file path
+        original.addPerson(IDA);
+        jsonDietLahStorage.saveDietLah(original); // file path not specified
+        readBack = jsonDietLahStorage.readDietLah().get(); // file path not specified
+        assertEquals(original, new DietLah(readBack));
+
+    }
+
+    @Test
+    public void saveAddressBook_nullAddressBook_throwsNullPointerException() {
```
Comment (ssoonwee) [inline src/test/java/seedu/address/storage/JsonDietLahStorageTest.java]: I think for some of these methods we have to rename from AddressBook also

File: src/test/java/seedu/address/storage/JsonDietLahStorageTest.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        try {
+            new JsonDietLahStorage(Paths.get(filePath))
+                    .saveDietLah(dietLah, addToTestDataPathIfNotNull(filePath));
+        } catch (IOException ioe) {
+            throw new AssertionError("There should not be an error writing to the file.", ioe);
+        }
+    }
+
+    @Test
+    public void saveAddressBook_nullFilePath_throwsNullPointerException() {
```
Comment (ssoonwee) [inline src/test/java/seedu/address/storage/JsonDietLahStorageTest.java]: I think for some of these methods we have to rename from AddressBook also

**Review by ssoonwee — CHANGES_REQUESTED** [review]
Some of the test methods require re-naming to dietLah! as indicated

**Review by ssoonwee — CHANGES_REQUESTED** [review]
Some of the test methods require re-naming to dietLah! as indicated

File: src/test/java/seedu/address/storage/JsonDietLahStorageTest.java (hunk 1 above)
Comment (sjq-sohjunqi) [reply src/test/java/seedu/address/storage/JsonDietLahStorageTest.java]: Ah yes, thanks for pointing it out!

**Comment by sjq-sohjunqi** [comment]
As per requested, I have renamed the test cases to Diet Lah as well as all other comments

File: src/test/java/seedu/address/storage/JsonDietLahStorageTest.java (hunk 2 above)
Comment (sjq-sohjunqi) [reply src/test/java/seedu/address/storage/JsonDietLahStorageTest.java]: Ah yes, thanks for pointing it out!

**Review by ssoonwee — APPROVED** [review]

**Pull request merged**